OnlineShop.Services.Catalog.Domain/Repositories/Interfaces/IProductRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using OnlineShop.Services.Catalog.Domain.Models.Data;
+
+namespace OnlineShop.Services.Catalog.Domain.Repositories.Interfaces
+{
+    public interface IProductRepository
+    {
+        Task<IEnumerable<Product>> GetAllAsync();
```

Comment:
Pagination needs to be implemented as fetching all data from the database significantly reduces performance

Reply:
I was planning to add this feature in future together with filtering